cmd/main.go

```diff
@@ -86,16 +90,58 @@ func main() {
 	// Register routes - no need for project and task handlers as they're handled via RPC
 	routes.RegisterRoutes(app, instanceHandler, userHandler, rpcHandler)
 
-	// Start server
-	port := os.Getenv("SERVER_PORT")
-	if port == "" {
-		port = "8080"
+	// Create context that listens for the interrupt signal from the OS.
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer stop()
+
+	// Create a WaitGroup to wait for goroutines to finish
+	var wg sync.WaitGroup
+
+	// Launch worker with the cancellable context and WaitGroup
+	wg.Add(1) // Increment counter before launching goroutine
+	go services.LaunchWorker(ctx, &wg, taskService)
+
+	// Start server in a goroutine so that it doesn't block.
+	var errChan = make(chan error)
+	go func() {
+		port := os.Getenv("SERVER_PORT")
+		if port == "" {
+			port = "8080"
+		}
+		log.Info("Server starting on :" + port)
+		if err := app.Listen(":" + port); err != nil {
+			// Send error to the channel
+			errChan <- err
+		}
+	}()
+
+	// Listen for the interrupt signal or the server error.
+	select {
+	case err := <-errChan:
+		log.Errorf("Server failed to start: %v", err)
+	case <-ctx.Done():
+		log.Info("Received interrupt signal, shutting down gracefully")
```

Comment:
[nit] This piece of code could have been moved to some helper func

Reply:
I don't think it can. It is blocking the program exit until one of the signals are hit. What would be the benefit of moving to a helper function?

Reply:
having a cleaner code